traits/src/currency.rs

```diff
@@ -643,6 +643,24 @@ impl<AccountId, CurrencyId, Balance> OnDust<AccountId, CurrencyId, Balance> for
 	fn on_dust(_: &AccountId, _: CurrencyId, _: Balance) {}
 }
 
+/// Handler for a newly created account
+pub trait OnNewTokenAccount<AccountId, CurrencyId> {
```

Comment:
I don't really like the idea of bunch different traits for each individual purposes.

I think we can just reuse `Happened<(&AccountId, CurrencyId)>` instead.

Otherwise we can have `trait OnAccountUpdated { fn updated }`

Reply:
Have a look. Not sure how I would get around the `who.clone()` induced by the change to `Happened`